nupic/research/TP.py

```diff
@@ -384,8 +384,6 @@ def __getstate__(self):
     for ephemeralMemberName in self._getEphemeralMembers():
       state.pop(ephemeralMemberName, None)
 
-    state['_random'] = self.getRandomState()
```

Comment:
This presents problems when I use capnp to de/serialize `Random` instance. I'm using the `RandomProto` schema in same way that it is used in `SpatialPoolerProto` instead of only store `seed` and create an instance of Random with this seed. 

Anyway, this lead to think that this causes a conflict with pickle. I tested only the testReadWrite test in an isolated way, certainly picke serialization will fail with this removal. I have to check how make these 2 serilization methods should work together.